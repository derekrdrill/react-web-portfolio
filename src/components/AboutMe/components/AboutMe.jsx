import React from 'react';
import styled, { createGlobalStyle } from 'styled-components';
import { Container, Row, Col } from 'react-bootstrap';
import { Typography, List, ListItem } from '@mui/material';

import { DarkLightModeContext } from '../../DarkLightMode/context/DarkLightModeContext';

import { ABOUT_ME_TEXT } from '../constants/ABOUT_ME_TEXT';
import bitmojiChillinWithBirds from '../../../assets/bitmoji_chillinWithBirds1.png';

export const AboutMe = ({ id }) => {
  // test change

  const { darkMode } = React.useContext(DarkLightModeContext);

  return (
    <>
      <PageBodyStyle darkMode={darkMode} />
      <AboutMeContainer darkMode={darkMode} id={id} fluid>
        <Row>
          <Col className='py-3'>
            <AboutMeTitle className='text-center' component='h1' darkMode={darkMode} variant='h3'>
              ABOUT ME
            </AboutMeTitle>
          </Col>
        </Row>
        <Row>
          <AboutMeTextContainer
            darkMode={darkMode}
            xs={{ span: 10, order: 'last', offset: 1 }}
            lg={{ span: 5, order: 'first', offset: 1 }}
          >
            <AboutMeText darkMode={darkMode} variant='subtitle1' component='p'>
              {ABOUT_ME_TEXT[1].text}
            </AboutMeText>
            <AboutMeText darkMode={darkMode} variant='subtitle1' component='p'>
              {ABOUT_ME_TEXT[2].text}
            </AboutMeText>
            <List style={{ color: 'white' }}>
              <AboutMeListItem darkMode={darkMode}>{ABOUT_ME_TEXT[3].text}</AboutMeListItem>
              <AboutMeListItem darkMode={darkMode}>{ABOUT_ME_TEXT[4].text}</AboutMeListItem>
              <AboutMeListItem darkMode={darkMode}>{ABOUT_ME_TEXT[5].text}</AboutMeListItem>
              <AboutMeListItem darkMode={darkMode}>{ABOUT_ME_TEXT[6].text}</AboutMeListItem>
            </List>
            <AboutMeText darkMode={darkMode} variant='subtitle1' component='p'>
              {ABOUT_ME_TEXT[7].text}
            </AboutMeText>
            <AboutMeText darkMode={darkMode} variant='subtitle1' component='p'>
              {ABOUT_ME_TEXT[8].text}
            </AboutMeText>
            <AboutMeText darkMode={darkMode} variant='subtitle1' component='p'>
              {ABOUT_ME_TEXT[9].text}
            </AboutMeText>
          </AboutMeTextContainer>
          <AboutMeBitmojiContainer xs={12} lg={6}>
            <img src={bitmojiChillinWithBirds} alt='Chillin with birds' />
          </AboutMeBitmojiContainer>
        </Row>
      </AboutMeContainer>
    </>
  );
};

const PageBodyStyle = createGlobalStyle(({ darkMode }) => ({
  body: {
    backgroundColor: darkMode ? '#030200' : 'whitesmoke',
  },
}));

export const AboutMeContainer = styled(Container)(({ darkMode }) => ({
  padding: '30px 50px 200px 50px',
  borderTop: '2px solid transparent',
  borderImage: darkMode
    ? 'linear-gradient(to right, rgba(248, 184, 255, 1), skyblue, gainsboro)'
    : 'linear-gradient(to right, rgba(176, 52, 197, 1), rgba(86, 206, 210, 1))',
  borderImageSlice: 1,
}));

export const AboutMeTitle = styled(Typography)(({ darkMode }) => ({
  fontFamily: 'Shizuru',
  fontWeight: darkMode ? 'normal' : 'bold',
  color: darkMode ? 'gainsboro' : 'darkslategray',
}));

export const AboutMeTextContainer = styled(Col)(({ darkMode }) => ({
  padding: 40,
  height: 550,
  overflowY: 'auto',
  border: '1px dashed #383838',
  borderRadius: 2,
  backgroundColor: darkMode ? '#303030' : '#eee5c1',
}));

export const AboutMeBitmojiContainer = styled(Col)({
  textAlign: 'center',
  paddingTop: 75,
  paddingBottom: 20,
});

export const AboutMeText = styled(Typography)(({ darkMode }) => ({
  fontFamily: 'Kufam',
  color: darkMode ? 'gainsboro' : '#333333',
  marginBottom: 15,
}));

export const AboutMeListItem = styled(ListItem)(({ darkMode }) => ({
  fontFamily: 'Kufam',
  color: darkMode ? 'gainsboro' : '#333333',
}));
