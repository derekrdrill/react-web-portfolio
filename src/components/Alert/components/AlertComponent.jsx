import React, { useContext } from 'react';
import styled, { css, keyframes } from 'styled-components';
import { Alert, AlertTitle } from '@mui/material';
import { AlertContext } from '../context/AlertContext';

export const AlertComponent = () => {
  const { alert, fadeOut } = useContext(AlertContext);

  return (
    alert && (
      <StyledAlert severity={alert.type} $fadeOut={fadeOut}>
        <AlertTitle>{alert.title}</AlertTitle>
        {alert.msg}
      </StyledAlert>
    )
  );
};

const fadeOut = keyframes({
  '0%': {
    opacity: 1,
  },
  '100%': {
    opacity: 0,
  },
});

const StyledAlert = styled(Alert)`
  animation-name: ${props =>
    props.$fadeOut
      ? css`
          ${fadeOut}
        `
      : 'none'};
  animation-duration: 500ms;
  animation-direction: linear;
`;
