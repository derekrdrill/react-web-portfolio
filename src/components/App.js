import React from 'react';

const AboutMe = React.lazy(() => import('./AboutMe/components/AboutMe'));
const ConnectWithMe = React.lazy(() => import('./ConnectWithMe/components/ConnectWithMe'));
const Homepage = React.lazy(() => import('./Homepage/Homepage'));
const Projects = React.lazy(() => import('./Projects/components/Projects'));

const App = () => (
  <div className='App'>
    <Homepage id='home' />
    <AboutMe id='about-me' />
    <Projects id='projects' />
    <ConnectWithMe id='connect-with-me' />
  </div>
);

export default App;
