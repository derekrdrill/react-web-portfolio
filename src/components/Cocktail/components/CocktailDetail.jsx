import React from 'react';
import styled from 'styled-components';
import { Grid, Typography } from '@mui/material';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faPlayCircle } from '@fortawesome/fontawesome-free-solid';

import { CocktailContext } from '../../Cocktail/context/CocktailContext';
import { DarkLightModeContext } from '../../DarkLightMode/context/DarkLightModeContext';

import { LoaderSpinner } from '../../LoaderSpinner/LoaderSpinner';

const CocktailDetail = () => {
  const { searchResults, searchResultsLength, loading } = React.useContext(CocktailContext);
  const { darkMode } = React.useContext(DarkLightModeContext);

  return (
    <>
      {loading && <LoaderSpinner open />}
      {!loading && searchResults && searchResults.length > 0 && (
        <CocktailDetailContainer container rowSpacing={2}>
          <Grid item xs={1} md={2} />
          <Grid item xs={10}>
            <Typography variant='h5'>
              {`${searchResultsLength} ${searchResultsLength === 1 ? 'drink' : 'drinks'} listed`}
            </Typography>
          </Grid>
          <Grid item xs={12} />
          <Grid item xs={12} />
          {searchResults.map((searchResult, searchResultKey) => (
            <>
              <CocktailResultsContainer key={searchResultKey} container rowSpacing={4}>
                <Grid item xs={1} md={2} />
                <Grid item xs={11} md={3}>
                  <CocktailImage src={searchResult.strDrinkThumb} />
                </Grid>
                <Grid item xs={1} sm={1} />
                <Grid item xs={10} md={5}>
                  <Typography variant='h3'>{searchResult.strDrink}</Typography>
                  <br />
                  <Grid container spacing={4}>
                    <Grid item xs={12} sm={6}>
                      <Typography variant='h5'>Ingredients</Typography>
                      {Object.keys(searchResult)
                        .filter(result => result.includes('strIngredient'))
                        .map(
                          (ingredient, ingredientKey) =>
                            searchResult[ingredient] && (
                              <Typography
                                key={`${searchResult.idDrink}${ingredientKey}`}
                                variant='subtitle2'
                              >
                                {`${searchResult[`strMeasure${ingredientKey + 1}`] ?? ''} ${
                                  searchResult[ingredient]
                                }`}
                              </Typography>
                            ),
                        )}
                    </Grid>
                    <Grid item xs={12} sm={6}>
                      <Typography variant='h5'>Glass</Typography>
                      <Typography variant='subtitle2'>{searchResult.strGlass}</Typography>
                    </Grid>
                  </Grid>
                  <br />
                  <Typography variant='h5'>Instructions</Typography>
                  <Typography variant='body2'>{searchResult.strInstructions}</Typography>
                  {searchResult.youtubeData && (
                    <CocktailVideoTutorialLinkContainer item xs={12}>
                      <CocktailVideoTutorialLink
                        href={searchResult.youtubeData.url}
                        target='_blank'
                        rel='noreferrer'
                      >
                        <CocktailVideoTutorial
                          className='youtube-img'
                          src={searchResult.youtubeData.bestThumbnail.url}
                        />
                        <CocktailVideoTutorialPlayButton icon={faPlayCircle} />
                        <CocktailVideoTutorialTitleContainer className='youtube-title'>
                          <CocktailVideoTutorialText variant='body1'>
                            {searchResult.youtubeData.title}
                          </CocktailVideoTutorialText>
                          <CocktailVideoTutorialText variant='subtitle2'>
                            {searchResult.youtubeData.duration}
                          </CocktailVideoTutorialText>
                        </CocktailVideoTutorialTitleContainer>
                      </CocktailVideoTutorialLink>
                    </CocktailVideoTutorialLinkContainer>
                  )}
                </Grid>
                <Grid xs={1} />
                <Grid xs={1} />
                <CocktailResultsDivider
                  item
                  xs={10}
                  darkMode={darkMode}
                  lastItem={searchResultKey + 1 === searchResultsLength}
                />
                <Grid xs={1} />
              </CocktailResultsContainer>
            </>
          ))}
        </CocktailDetailContainer>
      )}
    </>
  );
};

export default CocktailDetail;

export const CocktailDetailContainer = styled(Grid)({
  marginBottom: 150,
  position: 'relative',
  top: 100,
});

export const CocktailImage = styled.img({
  borderRadius: 30,
  height: 275,
  width: 320,
});

export const CocktailResultsContainer = styled(Grid)({
  padding: '40px 0',
});

export const CocktailResultsDivider = styled(Grid)(({ darkMode, lastItem }) => ({
  borderBottom: !lastItem && `solid 1px ${darkMode ? 'gainsboro' : 'grey'}`,
  margin: '0 30px',
  padding: '50px 0',
}));

export const CocktailVideoTutorial = styled.img({
  height: 215,
  width: 315,
});

export const CocktailVideoTutorialLinkContainer = styled(Grid)({
  ':hover': {
    '.youtube-img': {
      opacity: 0.4,
    },
    '.youtube-title': {
      visibility: 'visible',
    },
  },
  height: 215,
  transform: 'translateY(20px)',
  width: 315,
});

export const CocktailVideoTutorialLink = styled.a({
  cursor: 'pointer',
  textDecoration: 'none',
  zIndex: 2,
});

export const CocktailVideoTutorialText = styled(Typography)({
  color: 'white',
  fontWeigt: 'bold',
  zIndex: 0,
});

export const CocktailVideoTutorialTitleContainer = styled.div({
  maxWidth: 300,
  transform: 'translate(10px, -250px)',
  visibility: 'hidden',
});

export const CocktailVideoTutorialPlayButton = styled(FontAwesomeIcon)({
  color: 'red',
  height: 30,
  transform: 'translate(145px, -130px)',
  width: 30,
});
