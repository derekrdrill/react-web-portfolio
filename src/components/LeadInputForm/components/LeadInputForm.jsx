import React, { useState } from 'react';
import $ from 'jquery';
import styled from 'styled-components';
import { Container, Row, Col } from 'react-bootstrap';
import { Typography, Button } from '@mui/material';

import { DynamicFormInputs } from '../../DynamicFormInputs/DynamicFormInputs';
import { LoaderSpinner } from '../../LoaderSpinner/LoaderSpinner';

import { addLeadInput } from '../context/LeadInputActions';

import { LEAD_INFO_INPUTS } from '../constants/LEAD_INFO_INPUTS';

export const LeadInputForm = () => {
  const [leads, setLeads] = useState({});
  const [loading, setLoading] = useState(null);

  const updateLeadsObject = e => {
    let key = e.target.id;
    let value = e.target.value;
    let updatedValue = { [key]: value };

    setLeads(leads => ({
      ...leads,
      ...updatedValue,
    }));
  };

  const postApi = async () => {
    setLoading(true);
    setTimeout(() => {
      addLeadInput(leads);
      setLoading(null);
      $('input').val('');
    }, 1000);
  };

  return (
    <Container>
      <Row>
        <Col xs={{ span: 6, offset: 3 }} lg={{ span: 4, offset: 3 }} className='mt-5'>
          <Typography variant='h6' component='h1'>
            Please fill in your information below
          </Typography>
        </Col>
      </Row>
      <InputsRow>
        <Col xs={{ span: 12 }} lg={{ span: 6, offset: 3 }}>
          <DynamicFormInputs inputs={LEAD_INFO_INPUTS} onChange={updateLeadsObject} />
        </Col>
      </InputsRow>
      <SubmitButtonRow>
        <Col xs={{ span: 12 }} lg={{ span: 1, offset: 8 }}>
          <Button onClick={postApi} variant='contained' fullWidth>
            Submit
          </Button>
        </Col>
      </SubmitButtonRow>
      <LoaderSpinner open={loading} />
    </Container>
  );
};


const InputsRow = styled(Row)({
  marginTop: '5%',
});

const SubmitButtonRow = styled(Row)({
  marginTop: '5%',
});
