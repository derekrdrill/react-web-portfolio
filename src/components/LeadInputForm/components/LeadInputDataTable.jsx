import React from 'react';
import { DynamicDataTable } from '../../DynamicDataTable/components/DynamicDataTable';

export const headers = [
  { headerID: 'firstName', headerName: 'First Name' },
  { headerID: 'lastName', headerName: 'Last Name' },
  { headerID: 'email', headerName: 'Email' },
  { headerID: 'phone', headerName: 'Phone Number' },
];

const getLeadsAPICall = 'http://localhost:3001/leadInput';
const deleteLeadsAPICall = 'http://localhost:3001/deleteLeadInput';
const editLeadsAPICall = 'http://localhost:3001/replaceLeadInput';

export const LeadInputDataTable = () => (
  <DynamicDataTable
    checkAllColor='primary'
    checkOneColor='secondary'
    deleteDataRowsAPICall={deleteLeadsAPICall}
    deleteDataRowNameItems={['firstName', ' ', 'lastName']}
    editDataRowsAPICall={editLeadsAPICall}
    getDataRowsAPICall={getLeadsAPICall}
    headers={headers}
    size='small'
  />
);
