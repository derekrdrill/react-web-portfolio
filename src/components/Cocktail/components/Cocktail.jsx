import React from 'react';
import PropTypes from 'prop-types';
import styled, { createGlobalStyle } from 'styled-components';

import { AlertProvider } from '../../Alert/context/AlertContext';
import { DarkLightModeContext } from '../../DarkLightMode/context/DarkLightModeContext';

import CocktailDetail from './CocktailDetail';
import CocktailSearch from './CocktailSearch';

const Cocktail = () => {
  const { darkMode } = React.useContext(DarkLightModeContext);

  return (
    <AlertProvider>
      <PageStyle darkMode={darkMode} />
      <CocktailSearch />
      <CocktailDetail />
    </AlertProvider>
  );
};

Cocktail.propTypes = {};

export default Cocktail;

const PageStyle = createGlobalStyle(({ darkMode }) => ({
  body: {
    backgroundColor: darkMode ? '#1B0008' : '#FBAF93',
  },
  'h1, h2, h3, h4, h5, h6, p': {
    fontFamily: 'Trebuchet MS, sans-serif',
    fontWeight: 'bold',
    color: darkMode ? 'beige' : '#464646',
  },
}));
