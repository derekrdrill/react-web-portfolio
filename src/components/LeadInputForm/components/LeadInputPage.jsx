import React, { useContext } from 'react';
import styled, { createGlobalStyle } from 'styled-components';
import { Container, Row, Col } from 'react-bootstrap';
import { AppBar, Toolbar, Button, Tooltip, Typography } from '@mui/material';
import { LeadInputContext } from '../context/LeadInputContext';
import { LeadInputForm } from './LeadInputForm';
import { LeadInputDataTable } from './LeadInputDataTable';

export const LeadInputPage = () => {
  const { leadInputDispatch, page, tooltipOpen } = useContext(LeadInputContext);

  const handlePageButtonClick = () => {
    const newPage = page === 'form' ? 'table' : 'form';
    leadInputDispatch({ type: 'SWITCH_PAGES', payload: { page: newPage, tooltipOpen: false } });
  };

  const handleTooltipToggle = () => leadInputDispatch({ type: 'TOGGLE_TOOLTIP', payload: !tooltipOpen });

  return (
    <>
      <PageBodyStyle />
      <LeadInputPageHeader>
        <Toolbar>
          <Container fluid>
            <Row>
              <ButtonContainer xs={12}>
                <Tooltip
                  open={tooltipOpen}
                  onClose={handleTooltipToggle}
                  onOpen={page === 'form' && handleTooltipToggle}
                  title={
                    <Typography variant='subtitle2' component='p'>
                      This would likely be an admin button in a real-life scenario
                    </Typography>
                  }
                >
                  <Button onClick={handlePageButtonClick}>
                    {page === 'form' ? 'See all records' : 'Go back to input form'}
                  </Button>
                </Tooltip>
              </ButtonContainer>
            </Row>
          </Container>
        </Toolbar>
      </LeadInputPageHeader>
      <LeadInputPageContent>
        {page === 'form' && <LeadInputForm />}
        {page === 'table' && <LeadInputDataTable />}
      </LeadInputPageContent>
    </>
  );
};

const PageBodyStyle = createGlobalStyle({
  body: {
    backgroundColor: 'tan',
  },
});

const LeadInputPageHeader = styled(AppBar)({
  top: 80,
  boxShadow: 'none',
  backgroundColor: 'inherit',
});

const LeadInputPageContent = styled.div({
  marginTop: '10%',
});

const ButtonContainer = styled(Col)({
  textAlign: 'right',
});
