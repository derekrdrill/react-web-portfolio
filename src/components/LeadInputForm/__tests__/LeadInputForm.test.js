import 'regenerator-runtime/runtime';

import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';

import { LeadInputForm } from '../components/LeadInputForm';

const renderer = new ShallowRenderer();

describe('Lead Input Form tests', () => {
  it('renders correctly', () => {
    renderer.render(<LeadInputForm />);
    const result = renderer.getRenderOutput();
    expect(result).toMatchSnapshot();
  });
});
