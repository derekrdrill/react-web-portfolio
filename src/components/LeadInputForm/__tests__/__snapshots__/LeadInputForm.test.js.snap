// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Lead Input Form tests renders correctly 1`] = `
<div>
  <Memo(l) />
  <Container
    fluid={false}
  >
    <Row>
      <Col
        className="mt-5"
        lg={
          Object {
            "offset": 3,
            "span": 4,
          }
        }
        xs={
          Object {
            "offset": 3,
            "span": 6,
          }
        }
      >
        <ForwardRef(Typography)
          component="h1"
          variant="h6"
        >
          Please fill in your information below
        </ForwardRef(Typography)>
      </Col>
    </Row>
    <Styled(Row)>
      <Col
        lg={
          Object {
            "offset": 3,
            "span": 6,
          }
        }
        xs={
          Object {
            "span": 12,
          }
        }
      >
        <DynamicFormInputs
          inputs={
            Array [
              Object {
                "color": "primary",
                "fontFamily": "Kufam",
                "fullWidth": true,
                "id": "firstName",
                "inputBackgroundColor": "gainsboro",
                "inputPaddingBottom": 5,
                "inputborderRadius": 4,
                "label": "First Name",
                "labelColor": "#28282B",
                "variant": "filled",
                "xs": 12,
              },
              Object {
                "color": "primary",
                "fontFamily": "Kufam",
                "fullWidth": true,
                "id": "lastName",
                "inputBackgroundColor": "gainsboro",
                "inputPaddingBottom": 5,
                "inputborderRadius": 4,
                "label": "Last Name",
                "labelColor": "#28282B",
                "variant": "filled",
                "xs": 12,
              },
              Object {
                "color": "primary",
                "fontFamily": "Kufam",
                "fullWidth": true,
                "id": "email",
                "inputBackgroundColor": "gainsboro",
                "inputPaddingBottom": 5,
                "inputborderRadius": 4,
                "label": "Email",
                "labelColor": "#28282B",
                "variant": "filled",
                "xs": 12,
              },
              Object {
                "color": "primary",
                "fontFamily": "Kufam",
                "fullWidth": true,
                "id": "phone",
                "inputBackgroundColor": "gainsboro",
                "inputPaddingBottom": 5,
                "inputborderRadius": 4,
                "label": "Phone Number",
                "labelColor": "#28282B",
                "variant": "filled",
                "xs": 12,
              },
            ]
          }
          onChange={[Function]}
        />
      </Col>
    </Styled(Row)>
    <Styled(Row)>
      <Col
        lg={
          Object {
            "offset": 8,
            "span": 1,
          }
        }
        xs={
          Object {
            "span": 12,
          }
        }
      >
        <ForwardRef(Button)
          fullWidth={true}
          onClick={[Function]}
          variant="contained"
        >
          Submit
        </ForwardRef(Button)>
      </Col>
    </Styled(Row)>
    <LoaderSpinner
      open={null}
    />
  </Container>
</div>
`;
